filters/accesslog/control.go

```diff
@@ -87,3 +91,32 @@ func (*enableAccessLog) Name() string { return filters.EnableAccessLogName }
 func (al *enableAccessLog) CreateFilter(args []interface{}) (filters.Filter, error) {
 	return extractFilterValues(args, true)
 }
+
+type maskAccessLogQuery struct{}
+
+// NewMaskAccessLogQuery creates a filter spec to mask specific query parameters from the access log for a specific route.
+// Takes in query param keys as arguments. When provided, the value of these keys are masked (i.e., hashed).
+//
+//	maskAccessLogQuery("key_1", "key_2") to mask the value of provided keys in the access log.
+func NewMaskAccessLogQuery() filters.Spec {
+	return &maskAccessLogQuery{}
+}
+
+func (*maskAccessLogQuery) Name() string { return filters.MaskAccessLogQueryName }
+
+func (al *maskAccessLogQuery) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) <= 0 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	keys := make([]string, len(args))
+	for i := range args {
+		if key, ok := args[i].(string); ok && key != "" {
+			keys[i] = key
+		} else {
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	return &AccessLogFilter{Enable: true, MaskedQueryParams: keys}, nil
```

Comment:
This will override any preceding filter confgiration.

Maybe we should think about supporting a usecase where we could specify [default masked values](https://github.com/zalando-incubator/kubernetes-on-aws/blob/c78b2da5a3d0e9b59517f1d9317502369c347aae/cluster/config-defaults.yaml#L95)
```
disableAccessLog(2,3,404,429) -> accessLogMaskQueryParams("access_token") -> fifo(2000,20,"1s")
```
and then user may specify additional values.
OTOH proposed solution is simple and in line with what already exists.

Reply:
Hey @AlexanderYastrebov 

Yeah, I wanted to keep it simple. In hindsight, I am not sure my thinking is right.

If the current implementation is a no-go, I will try and spend a bit more time and see if I can come up with something when I get back from vacay (from tomorrow till the beginning of next month.)